animated_sprite.h

```diff
@@ -36,6 +36,8 @@ class AnimatedSprite : public Sprite {
   // Returns the current source location for the animated sprite.
   SDL_Rect GetSourceLocation() const override;
```

Comment:
GetSourceLocation is a bad name now, since "Location" is a different concept than "SDL_Rect".

Let's make this return an AnimationFrame anyways

The base class Sprite can just return an animation frame with offset 0, 0